crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -13,91 +13,107 @@
 // from outside) so it is only at that point that we need to check the hashes individually to the
 // find the one that got mined (if any).
 
-use super::{flashbots_api::FlashbotsApi, ESTIMATE_GAS_LIMIT_FACTOR};
+use super::ESTIMATE_GAS_LIMIT_FACTOR;
 use crate::settlement::Settlement;
 use anyhow::{anyhow, ensure, Context, Error, Result};
 use contracts::GPv2Settlement;
-use ethcontract::{transaction::Transaction, Account};
+use ethcontract::{contract::MethodBuilder, dyns::DynTransport, transaction::Transaction, Account};
 use futures::FutureExt;
 use gas_estimation::{EstimatedGasPrice, GasPriceEstimating};
 use primitive_types::{H256, U256};
 use shared::Web3;
-use std::time::{Duration, Instant, SystemTime};
+use std::time::{Duration, Instant};
 use web3::types::TransactionReceipt;
 
-pub struct FlashbotsSolutionSubmitter<'a> {
+/// Parameters for transaction submitting
+#[derive(Clone, Default)]
+pub struct SubmitterParams {
+    /// Desired duration to include the transaction in a block
+    pub target_confirm_time: Duration, //todo change to blocks in the following PR
+    /// Estimated gas consumption of a transaction
+    pub gas_estimate: U256,
+    /// Maximum max_fee_per_gas to pay for a transaction
+    pub gas_price_cap: f64,
+    /// Maximum duration of a single run loop
+    pub deadline: Option<Instant>,
+    /// Boost estimated gas price miner tip in order to increase the chances of a transaction being mined
+    pub additional_miner_tip: Option<f64>,
+    /// Resimulate and resend transaction on every retry_interval seconds
+    pub retry_interval: Duration,
+}
+
+pub struct TransactionHandle(pub String);
```

Comment:
One issue with associated types is that they make trait objects harder to use (you either need to specify the associated types upfront or some up with some wrapping type to make everything object safe).

Since both use 32 byte hashes, I suggest making it an `H256` which should cover all our bases for now, and then reevaluate when we find something that doesn't fit.

Reply:
@fleupold 
Is the comment equal to this one: https://github.com/gnosis/gp-v2-services/pull/1433#discussion_r757435096
@nlordell 
Makes sense, changed to H256